jxl_cli/src/main.rs

```diff
@@ -80,19 +85,267 @@ struct Opt {
     /// If specified, takes precedence over the bit depth in the input metadata
     #[clap(long)]
     override_bitdepth: Option<u32>,
+
+    #[clap(long, short, action)]
+    with_api: bool,
 }
 
-fn main() -> Result<(), Error> {
-    #[cfg(feature = "tracing-subscriber")]
-    {
-        use tracing_subscriber::{EnvFilter, fmt, prelude::*};
-        tracing_subscriber::registry()
-            .with(fmt::layer())
-            .with(EnvFilter::from_default_env())
-            .init();
+fn image_from_vec(vec: &[MaybeUninit<u8>], size: (usize, usize)) -> Result<Image<f32>> {
+    let mut image = Image::<f32>::new(size)?;
+    let mut rect = image.as_rect_mut();
+    for y in 0..size.1 {
+        let row = rect.row(y);
+        for x in 0..size.0 {
+            row[x] = f32::from_ne_bytes(unsafe {
```

Comment:
This needs _at minimum_ way more safety comments than it has.

I'd recommend instead using a 0-initialized `&[u8]`, at least for now.